crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -1869,3 +1858,37 @@ pub(crate) fn left_of_expr(t: &RExpr) -> Option<&RIdent> {
         _ => None,
     }
 }
+
+fn apply_intrinsics(intrinsics: &IntrinsicKind, raw: &Atom) -> Atom {
+    match intrinsics {
+        IntrinsicKind::Uppercase => raw.to_uppercase(),
+        IntrinsicKind::Lowercase => raw.to_lowercase(),
+        IntrinsicKind::Capitalize => {
+            if raw.is_empty() {
+                "".into()
+            } else {
+                let mut res = String::new();
+                let mut chars = raw.chars();
+
+                res.extend(chars.next().into_iter().flat_map(|v| v.to_uppercase()));
```

Comment:
Same here, and we don't need iterator here

Reply:
Sorry, I didn't understand exactly, 
isn't it necessary?
To make only the first letter Uppercase, there may be a way to import it as a mutable reference and modify it like this: `.get_mut(0..1).map(|prefix| prefix.to_uppercase())`.

Reply:
It's not necessary because it's an ASCII operation.
But my bad, I misread the variant.

Can you pre-allocate string above, and use `to_asscii_uppercase`?

Reply:
Looks like you are already assigning to `res` https://github.com/dudykr/stc/pull/295/commits/33b388860ccf2baf95bf70e1f999217e5bc190a1
Is it right to want a change like this?

Reply:
Yes